internal/api/http/services/expenses.go

```diff
@@ -0,0 +1,38 @@
+package services
+
+import (
+	"ExpensesTracker/pkg/api/http/requests"
+	"ExpensesTracker/pkg/api/http/responses"
+	"ExpensesTracker/pkg/database"
+	"ExpensesTracker/pkg/database/entities"
+	"errors"
+)
+
+// Source: https://stackoverflow.com/questions/39333102/how-to-create-or-update-a-record-with-gorm
+func UpdateExpense(summaryExpenseId uint, name string, updateSaving requests.CreateExpenseRequest) responses.ExpenseResponse {
```

Comment:
I would name it CreateOrUpdateExpense or UpsertExpense

Reply:
Done